pkg/dbmate/db.go

```diff
@@ -148,25 +153,41 @@ func (db *DB) Wait() error {
 	return db.wait(drv)
 }
 
-// CreateAndMigrate creates the database (if necessary) and runs migrations
+// CreateAndMigrate creates the database (if necessary), load schema.sql file
+// to the database (if autoload enabled) and then apply all pending migrations
 func (db *DB) CreateAndMigrate() error {
 	drv, err := db.Driver()
 	if err != nil {
 		return err
 	}
 
-	// create database if it does not already exist
-	// skip this step if we cannot determine status
-	// (e.g. user does not have list database permission)
-	exists, err := drv.DatabaseExists()
-	if err == nil && !exists {
+	// create database if it does not already exist skip this step if we cannot
+	// determine status (e.g. user does not have list database permission)
+	databaseExists, err := drv.DatabaseExists()
+	if err == nil && !databaseExists {
 		if err := drv.CreateDatabase(); err != nil {
 			return err
 		}
 	}
 
-	// migrate
-	return db.Migrate()
+	sqlDB, err := drv.Open()
+	if err != nil {
+		return err
+	}
+	defer dbutil.MustClose(sqlDB)
+
+	// if autoload enabled and migration table does not exist load schema.sql
+	if db.AutoLoadSchema {
+		migrationsTableExists, err := drv.MigrationsTableExists(sqlDB)
+		if err == nil && !migrationsTableExists {
+			if err := db.loadSchema(sqlDB); err != nil {
+				return err
+			}
+		}
+	}
+
+	// apply any pending migrations
+	return db.migrate()
```

Comment:
I don't understand why you couldn't still just invoke `db.Migrate()` here, and why you split it into `db.migrate()` and `db.Migrate()` just to move the `len(migrations) == 0` test, at the cost of forcing every call to `db.Migrate()` to do double the work of  `db.FindMigrations()`, once in `db.Migrate()` and a second time in `db.mgirate()`.  This feels like a mistake.

Reply:
You're right, there was a mistake